rotkehlchen/exchanges/coinbase.py

```diff
@@ -265,18 +314,17 @@ def edit_exchange_credentials(self, credentials: ExchangeAuthCredentials) -> boo
         changed = super().edit_exchange_credentials(credentials)
         if credentials.api_key is not None:
             try:
-                new_is_legacy = self.is_legacy_key(credentials.api_key)
-            except AuthenticationError as e:
-                log.error(f'Invalid coinbase API key format: {e}')
-                new_is_legacy = True
+                if (new_key_type := CoinbaseKeyType.detect_type(credentials.api_key)) != self.key_type:  # noqa: E501
+                    self.key_type = new_key_type
 
-            if new_is_legacy != self.is_legacy_api_key:  # Key type has changed
-                self.is_legacy_api_key = new_is_legacy
+                if self.key_type.is_legacy:
+                    self.session.headers.update({'CB-ACCESS-KEY': credentials.api_key})
+                else:
+                    self.api_key = credentials.api_key
```

Comment:
```suggestion
                if self.key_type.is_legacy:
                    self.session.headers.update({'CB-ACCESS-KEY': credentials.api_key})
                
                self.api_key = credentials.api_key
```
should self.api_key get set even if its a legacy key?